packages/graphql-mocks/src/graphql/handler.ts

```diff
@@ -72,25 +77,32 @@ export class GraphQLHandler {
       packOptions,
     });
 
+    const { resolverMap } = this;
+    if (!resolverMap) {
+      throw new Error('`pack` must be ran to create a final resolverMap');
+    }
+
     return graphql({
       ...graphqlArgs,
       source: query,
       schema,
       variableValues,
       contextValue,
+      fieldResolver: fieldResolverRouter(resolverMap),
+      typeResolver: typeResolverRouter(resolverMap),
     }) as ExecutionResult<DataResult>;
   }
 
-  protected async pack(): Promise<void> {
+  async pack(): Promise<void> {
     const { initialResolverMap, middlewares, packOptions, graphqlSchema, scalarMap } = this;
 
     if (!this.packed) {
       const { resolverMap, state } = await pack(initialResolverMap, middlewares, packOptions);
       this.state = state;
-      attachResolversToSchema(graphqlSchema, resolverMap);
```

Comment:
Replacing this with a resolver router doesn't appear to impact the `pack` time of installing the resolvers on the schema instance but it does mean that a schema is more easily re-used (except for attached custom scalars) by not modifying the schema but letting the schema fallback to the passed in `fieldResolver` and `typeResolver`, so this change is probably still worth keeping but not for performance.